### PR TITLE
tls/x509utils: introduce CertPool and CertPoolWriter interfaces

### PR DESCRIPTION
**Pull request by amery**
## Summary by CodeRabbit

- **New Features**
	- Introduced new interfaces for enhanced certificate management: `CertPool` for accessing x509 certificates and `CertPoolWriter` for modifying the certificate pool.
	- Added methods for retrieving, iterating, cloning, and exporting certificates, as well as adding, removing, and importing certificates.

### CODE REVIEW

**Comment by amery** [comment]
@CodiumAI-Agent /review

File: tls/x509utils/types.go [hunk 1]
```diff
@@ -28,3 +30,25 @@ type PrivateKey interface {
...
+// access to them which the standard [x509.CertPool] doesn't.
+type CertPool interface {
+	Get(ctx context.Context, name string) (*x509.Certificate, error)
+	ForEach(ctx context.Context, fn func(context.Context, *x509.Certificate) bool)
+
+	Clone() CertPool
+	Export() *x509.CertPool
+}
+
+// A CertPoolWrite extends the CertPool interface with write capabilities.
```
[the remark this answers is not in the document]
Comment (amery) [reply tls/x509utils/types.go:44]: uhm, `make tidy` seems broken

**Comment by amery** [comment]
rebased

**Comment by amery** [comment]
Thoughts regarding renaming Put() to Add()?

**Comment by karasz** [comment]
> Thoughts regarding renaming Put() to Add()?
nope, Put() is more fitting I think.

**Review by karasz — APPROVED** [review]
LGTM

**Pull request merged**